c_glib/arrow-glib/file-system.h

```diff
@@ -0,0 +1,236 @@
+                                        gsize n_paths,
+                                        GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_system_move(GArrowFileSystem *file_system,
+                                 const gchar *src,
+                                 const gchar *dest,
+                                 GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_system_copy_file(GArrowFileSystem *file_system,
+                                      const gchar *src,
+                                      const gchar *dest,
+                                      GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+GArrowInputStream *
+garrow_file_system_open_input_stream(GArrowFileSystem *file_system,
+                                     const gchar *path,
+                                     GError **error);
+
+/* TODO:
+GARROW_AVAILABLE_IN_1_0
+GArrowRandomAccessFile *
+garrow_file_system_open_input_file(GArrowFileSystem *file_system,
+                                   const gchar *path,
+                                   GError **error);
+*/
+
+GARROW_AVAILABLE_IN_1_0
+GArrowOutputStream *
+garrow_file_system_open_output_stream(GArrowFileSystem *file_system,
+                                      const gchar *path,
+                                      GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+GArrowOutputStream *
+garrow_file_system_open_append_stream(GArrowFileSystem *file_system,
+                                      const gchar *path,
+                                      GError **error);
+
+/* arrow::fs::SubTreeFileSystem */
+
+#define GARROW_TYPE_SUB_TREE_FILE_SYSTEM (garrow_sub_tree_file_system_get_type())
+G_DECLARE_DERIVABLE_TYPE(GArrowSubTreeFileSystem,
+                         garrow_sub_tree_file_system,
+                         GARROW,
+                         SUB_TREE_FILE_SYSTEM,
+                         GArrowFileSystem)
+struct _GArrowSubTreeFileSystemClass
+{
+  GArrowFileSystemClass parent_class;
+};
+
+GARROW_AVAILABLE_IN_1_0
+GArrowSubTreeFileSystem *
+garrow_sub_tree_file_system_new(const gchar *base_path,
+                                GArrowFileSystem *base_file_system);
+
+/* TODO: arrow::fs::SlowFileSystem */
+
+#define GARROW_TYPE_SLOW_FILE_SYSTEM (garrow_slow_file_system_get_type())
+G_DECLARE_DERIVABLE_TYPE(GArrowSlowFileSystem,
+                         garrow_slow_file_system,
+                         GARROW,
+                         SLOW_FILE_SYSTEM,
+                         GArrowFileSystem)
+struct _GArrowSlowFileSystemClass
+{
+  GArrowFileSystemClass parent_class;
+};
+
+/* TODO: GArrowLatencyGenerator
+GARROW_AVAILABLE_IN_1_0
+GArrowSlowFileSystem *
+garrow_slow_file_system_new(GArrowFileSystem *base_file_system,
+                            GArrowLatencyGenerator *latencies);
+*/
+
+GARROW_AVAILABLE_IN_1_0
+GArrowSlowFileSystem *
+garrow_slow_file_system_new_by_average_latency(GArrowFileSystem *base_file_system,
```

Comment:
I think it's better to use`garrow_slow_file_system_new_average_latency()` for the same reason as https://github.com/apache/arrow/pull/6105#discussion_r361770906.